java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightProducer.java

```diff
@@ -77,7 +77,9 @@ default PollInfo pollFlightInfo(CallContext context, FlightDescriptor descriptor
    */
   default SchemaResult getSchema(CallContext context, FlightDescriptor descriptor) {
     FlightInfo info = getFlightInfo(context, descriptor);
-    return new SchemaResult(info.getSchema());
+    return new SchemaResult(info
+            .getSchemaOptional()
+            .orElseThrow(() -> new IllegalArgumentException("No schema is present in FlightInfo")));
```

Comment:
It would be better to throw `CallStatus.INVALID_ARGUMENT.withDescription(...).asRuntimeException()`